ironfish/src/utils/array.test.ts

```diff
@@ -0,0 +1,50 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { Assert } from '../assert'
+import { ArrayUtils } from './array'
+
+describe('ArrayUtils', () => {
+  it('shuffles array', () => {
+    const items: number[] = []
+    for (let i = 0; i <= 10000; ++i) {
+      items.push(i)
+    }
+
+    const shuffled = ArrayUtils.shuffle(items)
+    expect(shuffled).not.toEqual(items)
+    expect(shuffled.sort()).toEqual(items)
```

Comment:
JS sorting uses string representations by default

```suggestion
    expect(shuffled.sort((a, b) => a - b).toEqual(items)
```